src/components/spreadsheet/table-wrapper.js

```diff
@@ -0,0 +1,686 @@
+                        props.studyUuid,
+                        props.currentNode?.id,
+                        groovyCr
+                    );
+            }
+        },
+        [props.currentNode?.id, props.studyUuid]
+    );
+
+    const validateEdit = useCallback(
+        (params) => {
+            // TODO: generic groovy updates should be replaced by specific hypothesis creations, like modifyLoad() below
+            // TODO: when no more groovy, remove changeCmd everywhere, remove requestNetworkChange()
+            let groovyCr =
+                'equipment = network.' +
+                TABLES_DEFINITION_INDEXES.get(tabIndex).groovyEquipmentGetter +
+                "('" +
+                params.data.id.replace(/'/g, "\\'") +
+                "')\n";
+
+            const wrappedEditedData = {
+                data: editingData,
+            };
+            Object.entries(priorValuesBuffer).forEach(([field, value]) => {
+                const column = gridRef.current.columnApi.getColumn(field);
+                const val = column.colDef.valueGetter
+                    ? column.colDef.valueGetter(wrappedEditedData)
+                    : editingData[field];
+
+                groovyCr +=
+                    column.colDef.changeCmd?.replace(/\{\}/g, val) + '\n';
+            });
+
+            console.log(groovyCr);
+            const editPromise = buildEditPromise(editingData, groovyCr);
+            Promise.resolve(editPromise)
+                .then(() => {
+                    const transaction = {
+                        update: [editingData],
+                    };
+                    gridRef.current.api.applyTransaction(transaction);
+                    setEditingData();
+                    setIsValidatingData(false);
+                })
+                .catch((promiseErrorMsg) => {
+                    console.error(promiseErrorMsg);
+                    rollbackEdit();
+                    let message = intl.formatMessage({
+                        id: 'paramsChangingDenied',
+                    });
+                    snackError({
+                        messageTxt: message,
+                        headerId: 'paramsChangingError',
+                    });
+                });
+        },
+        [
+            buildEditPromise,
+            editingData,
+            intl,
+            priorValuesBuffer,
+            rollbackEdit,
+            snackError,
+            tabIndex,
+        ]
+    );
+
+    //this listener is called for each cell modified
+    const handleCellEditing = useCallback(
+        (params) => {
+            addDataToBuffer(params.colDef.field, params.oldValue);
+        },
+        [addDataToBuffer]
+    );
+
+    //this listener is called once all cells listener have been called
+    const handleRowEditing = useCallback(
+        (params) => {
+            if (isValidatingData) {
+                //we call stopEditing again to prevent editors flickering
```

Comment:
comment and commented code seem to be in contradiction

Reply:
It was commented because it wasn't working as intented. I finally corrected the flickering issue it by adding the following test in `EditingCellRenderer` :
`if (!props.isValidatingData) {
props.startEditing();
}`

Since `EditingCellRenderer` was rerendering during the validation process, editors were opening up unintendedly